src/main/java/org/jenkinsci/plugins/parallel_test_executor/ParallelTestExecutor.java

```diff
@@ -155,14 +158,41 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListen
         return true;
     }
 
+    public static Map<String, TestClass>  findTestResultsInDirectory(Run<?,?> build, TaskListener listener, FilePath workspace){
+        if(workspace==null){
+            return Collections.emptyMap();
+        }
+        String[] tests = null;
+        Map<String, TestClass> data = new TreeMap<String, TestClass>();
+        final String baseDir = workspace.getRemote();
+        try {
+            tests = workspace.act(new MasterToSlaveCallable<String[], Throwable>() {
+                @Override
+                public String[] call() throws Throwable {
+
+                    return Util.createFileSet(new File(baseDir), "**/src/test/java/**/*Test.java").getDirectoryScanner().getIncludedFiles();
+                }
+            });
+        } catch (Throwable throwable) {
+            throwable.printStackTrace(listener.getLogger());
+            return data;
+        }
+        for(String test : tests){
+            test = test.split("src/test/java/")[1];
```

Comment:
does this work when running on windows?  (it would appear as though it does not)

Reply:
[This](https://ci.jenkins.io/job/Plugins/job/parallel-test-executor-plugin/job/PR-34/1/testReport/junit/org.jenkinsci.plugins.parallel_test_executor/ParallelTestExecutorUnitTest/findTestInJavaProjectDirectory/) suggests not:

```
java.lang.ArrayIndexOutOfBoundsException: 1
	at org.jenkinsci.plugins.parallel_test_executor.ParallelTestExecutor.findTestResultsInDirectory(ParallelTestExecutor.java:181)
	at org.jenkinsci.plugins.parallel_test_executor.ParallelTestExecutor.findTestSplits(ParallelTestExecutor.java:214)
	at org.jenkinsci.plugins.parallel_test_executor.ParallelTestExecutorUnitTest.findTestInJavaProjectDirectory(ParallelTestExecutorUnitTest.java:119)
```

Reply:
I did not test it on all platforms. I guess that in case of unix system it is alright to test it only for linux. I will run tests for Windows and fix bug. Thanks.